src/Compilers/Core/Portable/InternalUtilities/TextKeyedCache.cs

```diff
@@ -104,11 +108,12 @@ public void Free()
             // Array.Clear(this.localTable, 0, this.localTable.Length);
             // Array.Clear(sharedTable, 0, sharedTable.Length);
 
-            _pool.Free(this);
+            _pool?.Free(this);
```

Comment:
>_pool?.Free(this) [](start = 12, length = 17)

Callers should only invoke `Free()` for instances obtained from a pool.

Reply:
Indeed. The code base has been inconsistent on how it handles this case. The majority seemed to prefer using ?.Free here hence I've been going with that. 

I'm also okay with adding a Debug.Assert call here to eliminate the nullable warning.  

---
In reply to: [356900168](https://github.com/dotnet/roslyn/pull/40268#discussion_r356900168) [](ancestors = 356900168)